src/test/test-locale-util.c

```diff
@@ -24,10 +24,8 @@ static void test_get_locales(void) {
 static void test_locale_is_valid(void) {
         log_info("/* %s */", __func__);
 
-        assert_se(locale_is_valid("en_EN.utf8"));
```

Comment:
Dunno, maybe just hardcode the test. Any particular locale might not exist on a system.
```C
r = locale_is_valid("en_EN.utf8");
if (access("/usr/lib/locale/en_US.utf8") >= 0 || access("/usr/lib/locale/en_US.UTF-8") >= 0)
       assert_se(r == 1);
else
       assert_se(r >= 0);
```
This test will only work some of the time, but it'll work a lot of the time.

Reply:
This is nuts and there is no point in testing `newlocale()` like that.

Reply:
well, i think there's value in checking whether our expectations of the underlying OS primitives matches reality. Sure, glibc should have tests for newlocale(), but we build on that, and we should verify that what we build from it matches what we think it should work like.